src/Temporalio/Worker/WorkflowInstance.cs

```diff
@@ -935,6 +1034,403 @@ public override Task DelayAsync(DelayAsyncInput input)
                     }
                 });
             }
+
+            /// <inheritdoc />
+            public override Task<TResult> ScheduleActivityAsync<TResult>(
+                ScheduleActivityInput input)
+            {
+                if (input.Options.StartToCloseTimeout == null &&
+                    input.Options.ScheduleToCloseTimeout == null)
+                {
+                    throw new ArgumentException("Activity options must have StartToCloseTimeout or ScheduleToCloseTimeout");
+                }
+
+                return ExecuteActivityInternalAsync<TResult>(
+                    local: false,
+                    doBackoff =>
+                    {
+                        var seq = ++instance.activityCounter;
+                        var cmd = new ScheduleActivity()
+                        {
+                            Seq = seq,
+                            ActivityId = input.Options.ActivityID ?? seq.ToString(),
+                            ActivityType = input.Activity,
+                            TaskQueue = input.Options.TaskQueue ?? instance.Info.TaskQueue,
+                            Arguments = { instance.payloadConverter.ToPayloads(input.Args) },
+                            RetryPolicy = input.Options.RetryPolicy?.ToProto(),
+                            CancellationType = (Bridge.Api.WorkflowCommands.ActivityCancellationType)input.Options.CancellationType,
+                        };
+                        if (input.Headers is IDictionary<string, Payload> headers)
+                        {
+                            cmd.Headers.Add(headers);
+                        }
+                        if (input.Options.ScheduleToCloseTimeout is TimeSpan schedToClose)
```

Comment:
Why do these (and other places) have these `is` checks? Can't they only be of that type? Some particular reason to prefer this over a "not null" kinda check?

Reply:
> Some particular reason to prefer this over a "not null" kinda check?
Yes, it's only for nullness check. With this, `schedToClose` is a known non-null variable whereas `input.Options.ScheduleToCloseTimeout != null` does not provide that guarantee at compile time making things a bit ugly. But either works.